sections/errorhandling/correlationid.md

```diff
@@ -0,0 +1,35 @@
+# Correlation ID: help your logs tell you a story and give your error a context
+
+### One Paragraph Explainer
+
+Correlation ID is one of the best problem-solving patterns. It lets you link log records, even if they belong to different services. If your system consumes other services and is itself a producer service, adding a correlaction ID is a must. By this pattern, your transaction logs can become into a story that tells itself by filtering your logs with a specific correlation ID, instead of try linking the cross-services transaction logs to each other by yourself. It can save your day when a process including 20 different microservices throws an exception in one of them, and you have no idea where did the problem start across the flow.
+
+<br/>
+
+### Code Example: passing the correlation ID between services on the requet http context
+Here is an example of using [express-http-context](https://www.npmjs.com/package/express-http-context) library as part of a middleware to set the forwarded correlation ID on the http context after it was extracted from the request. 
+If the correlation ID doesn't exist on the request, since this may be the first request in the requests chain, it's being generated.
+
+```javascript
+const httpContext = require('express-http-context');
+
+app.use((req, res, next) => {
+  // Extract the correlation ID from the previous request, or generate it if this is the first request in the transaction
+  const correlationId = req.get('X-Correlation-ID') || uuid.v4();
+  
+  // Set the correaltion ID on the http context
+  httpContext.set('correlationId', correlationId);
+  
+  // Set the correaltion ID on the response
+  res.set('X-Correlation-ID', correlationId);
+
+  next();
+});
+```
+<br/>
+
+### Blog Quote: "The notion of a Correlation ID is simple. It’s a value that is common to all requests, messages and responses in a given transaction. With this simplicity you get a lot of power."
```

Comment:
Just an idea - I would put here an image of some log aggregator UI (e.g. ELK) that shows how the 'story' looks like. I can get one

Reply:
I've just found in our logs this example- 
https://drive.google.com/file/d/1AYdmHhSZGcJaGIFA1pdMX15NMwG-R3de/view?usp=sharing
(microservices real names and data are edited)
Is it the kind of story to show in an image you thought about?

Reply:
I think showing images could be great to help people understand the concept better!

Do you think you could also generate an image with unrelated logs? To show the before and after?

I think some cropping around the log area would help the reader to focus

Reply:
Yes, I also like @kevynb idea of showing an 'Anti-pattern' image that shows a messy log